src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -222,6 +222,22 @@ public void Register(Type t, DeepCopier cop, Serializer ser, Deserializer deser)
             Register(t, cop, ser, deser, false);
         }
 
+        private object InitializeSerializer(Type serializerType)
+        {
+            if (this.ServiceProvider == null)
+                return null;
+
+            if (!serializerType.GetCustomAttributes(typeof(SerializerAttribute), true).Any())
```

Comment:
Worth adding a comment here, like:
`// If the type lacks a Serializer attribute then it's a self-serializing type and all serialization methods must be static.`